apps/accelerate/chatllama/chatllama/rlhf/actor.py

```diff
@@ -615,77 +296,126 @@ def train(
-                        input_tokenized_id,
-                        input_tokenized_mask,
-                    )
+                    # tokenize the input
+                    if self.accelerate_enable:
+                        input_tokenized = self.model.module.tokenizer(
+                            input_text,
+                            return_tensors="pt",
+                            truncation=True,
+                            padding=True,
+                        )
+                    else:
+                        input_tokenized = self.model.tokenizer(
+                            input_text,
+                            return_tensors="pt",
+                            truncation=True,
+                            padding=True,
+                            max_length=self.config.max_sequence_length,
+                        )
+
+                        # split tokens and mask
+                        input_tokenized_id = input_tokenized["input_ids"]
+                        input_tokenized_mask = input_tokenized[
+                            "attention_mask"
+                        ]
+
+                        # add eos token at the end of the sequence
+                        (
+                            input_tokenized_id,
+                            input_tokenized_mask,
+                        ) = self.add_eos_token(
+                            input_tokenized_id,
+                            input_tokenized_mask,
+                        )
 
                     # split into input and output
                     training_output = input_tokenized_id[:, 1:]
                     training_input = input_tokenized_id[:, :-1]
                     attention_mask = input_tokenized_mask[:, :-1]
 
                     # move to device
-                    training_output = training_output.to(device)
-                    training_input = training_input.to(device)
-                    attention_mask = attention_mask.to(device)
+                    if self.config.load_8bit is False:
+                        training_output = training_output.to(self.device)
+                        training_input = training_input.to(self.device)
+                        attention_mask = attention_mask.to(self.device)
 
                 # forward pass
-                if self.config.deepspeed_enable:
+                if self.deepspeed_enable:
+                    # deepspeed
                     est_output = self.model_engine(
                         training_input, attention_mask
                     )
+                elif self.accelerate_enable:
+                    # accelerate
+                    est_output = self.model(training_input, attention_mask)
                 else:
-                    est_output = self.actor(training_input, attention_mask)
+                    # pytorch mixed precison
+                    with torch.autocast(
+                        device_type=self.config.device_type,
+                        dtype=torch.float16,
```

Comment:
just following documentation...
https://pytorch.org/docs/stable/notes/amp_examples.html
wrt to casting manually the tensors, this is better with less problem with types in the embedding.
It is not a config param because if you do not use fp16 you would use fp32 and is probably worse. 
not seen the point of adding the option for fp32.

Reply:
But what if I want to train the model in fp32 precision? (DeepSpeed for instance allows the user to select the precision)